design/specs/data-type-options.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Data Type Options
+description: 
+published: true
+date: 2022-09-01
+tags: 
+editor: markdown
+dateCreated: 2022-09-01
+---
+
+## Context
+
+Data types are used to define the type of data that a column can hold. Each data type has a set of options that can be configured to define how the data is interpreted and displayed. These set of options are split into two categories:
+
+- **Database**: These options are used to define how the data is stored in the database.
+- **Format**: Defines how the data is displayed to the user.
+
+## Problem
+
+The current implementation uses tabs to switch between the two categories of options. This is not ideal because data type options are used in Table Inspector, where tabs are already used to switch between different sections. We are trying to avoid having a tabbed interface within a tabbed interface.
+
+## Vertical Space Considerations
+
+The solution should consider the challenges in displaying the options in a small vertical space, such as the dropdown menu. Scroll bars should be avoided if possible.
+
+## Proposed Solution
+
+### Replace tabs with stacked sections
+
+Update Database and Format options to be stacked sections inside the Data Type section. The user should be able to access the options without the need of tabs. The styling should make the distinction between sections and sub-sections clear.
+
+### Use a dropdown menu for the data type selection instead of the scrollable list
+
+![image](/assets/design/specs/data-type-options/gL37Da5Tdp4iTZScWJPGC.png)
+
+### Align some of the label and field groups inline
+
+Inline alignment can be used for groups of fields in order to save vertical space.
+
+![image](/assets/design/specs/data-type-options/6VPzH8Cd7USbhf6jZpzQvc.png)
+
+## Alternatives
+
+### Remove the sub-sections and instead use an icon next to database-level options
+
+A warning message can be displayed inline when the user tries to change a database-level option. The icon can then be used for other instances where the user needs to be aware of database-level options.
+
+Removing sub-sections will make the UI cleaner.
+
+![image](/assets/design/specs/data-type-options/fpnLR7JtkJJZXGybZoYL8d.png)
```

Comment:
For both the proposed solution and the alternative:

We always need an explicit Save button. This save button _needs to be common_ for all the database options combined and _**not** per input_ because the determination of column type depends on the fully filled database options form. The behaviour of the save button and it's placement needs to be mentioned clearly on the spec.

Reply:
@pavish we have a separate spec for inline saving. I will include a reference here and have a wireframe for reference.

Reply:
I understand we have a separate spec for inline saving, my question is whether it will cover the entire form?

> we have a separate spec for inline saving

The current inline saving seems to be per input, each input having it's own save button. But for the data type menu, we cannot do that, because the type is calculated based on the result of all the related inputs. So, we need a common save button.

Reply:
@pavish yes, the entire section should be saved at once. I will include the example in the spec.

Reply:
@pavish I'm assuming that save is only required for the database options and that we won't be asking the user to save format options. Is that the right assumption?

Reply:
Yes, we can make format options save automatically. However, **I'd prefer the save button to save both database and format options together**. We show the same options in both data import menu and the table inspector, and it would be a lot simpler implementation-wise and maybe more user friendly, if they were as similar to each other as possible.

Reply:
It makes sense. I was concerned about the inconsistency between this and inline saving but I think you make a good point. I will update.

Reply:
I have further comments placed here: https://github.com/centerofci/mathesar-wiki/pull/59#pullrequestreview-1098147017